src/app/report/edit-display/edit-display.component.ts

```diff
@@ -107,30 +116,37 @@ export class EditDisplayComponent {
   }
 
   downloadReport(exportBinary: boolean, exportXML: boolean): void {
-    let queryString: string = this.report.xml ? this.report.storageId.toString() : this.report.uid.split('#')[0];
-    this.helperService.download(queryString + '&', this.currentView.storageName, exportBinary, exportXML);
+    const queryString: string = this.report.xml ? this.report.storageId.toString() : this.report.uid.split('#')[0];
+    this.helperService.download(`${queryString}&`, this.currentView.storageName, exportBinary, exportXML);
     this.httpService.handleSuccess('Report Downloaded!');
   }
 
   selectStubStrategy(event: Event): void {
-    this.saveChanges(true, (event.target as HTMLOptionElement).value);
+    this.saveChanges((event.target as HTMLOptionElement).value);
   }
 
   openDifferenceModal(type: ChangesAction): void {
     const reportDifferences: ReportDifference[] = [];
     if (this.report.xml && this.editFormComponent) {
-      reportDifferences.push(
-        this.getDifference('name', this.report.name, this.editFormComponent.name),
-        this.getDifference('description', this.report.description, this.editFormComponent.description),
-        this.getDifference('path', this.report.path, this.editFormComponent.path),
-        this.getDifference('transformation', this.report.transformation, this.editFormComponent.transformation),
-        this.getDifference('variables', this.report.variables, this.editFormComponent.variables),
-      );
-    } else {
+      reportDifferences.push(...this.getDifferences());
```

Comment:
These are two arrays, right, can't it be like this?
```suggestion
      reportDifferences + this.getDifferences();
```

Reply:
That's possible but that means changing `reportDifferences` to a let. I'd say having a const is better.

Reply:
Functionally that might be true, but performance wise it is asking more to do a spread and then add it again.

Reply:
Fixed it